### PR TITLE
Improve the git instructions to update a PR.

### LINKED ISSUE #12337: Contributor Guide git push after rebase needs -f
The last `git push` at the bottom of https://docs.docker.com/project/work-issue/#pull-and-rebase-frequently will fail, I think, unless it has `-f` added.

This is because the tip of the current branch will not match the remote repository, so Git will reject the update.

### PR DESCRIPTION
**Pull request by draghuram**
Signed-off-by: Raghuram Devarakonda draghuram@gmail.com

### CODE REVIEW

**Comment by bboreham** [comment]
I disagree that it closes #12337, because it is about a different file.

**Comment by thaJeztah** [comment]
@bboreham @draghuram agreed; this is about a different file, so I removed the `fixes xxx` from the comment.

**Comment by draghuram** [comment]
@bboreham You are right that my PR tackles different file. Sorry for the confusion. I pushed the commit again after removing the "Closing ..." line from the comment.

File: docs/sources/project/review-pr.md [hunk 1]
```diff
@@ -49,9 +49,12 @@ need to update your pull request with additional changes.
 
 To update your existing pull request:
 
-1. Change one or more files in your local `docker-fork` repository.
+1. Change one or more files in your local `docker-fork` repository
```
Comment (moxiegirl) [inline docs/sources/project/review-pr.md]: Thank you for the good catch.  The text has some issues.  Parentheticals as used here can indicate problems in  structure particularly in designing step text. These are being used to include system results and extra steps.  They need to extracted out, clarified, and positioned correctly.

--> Markdown: https://gist.github.com/moxiegirl/7bde985ae88f91ce36a2

To update your existing pull request:

1. Checkout the PR branch in your local `docker-fork` repository.

  This is the branch associated with your request.
	
2. Change a file and then stage it.

  The command syntax is:
 
       git add <filename>

3. Commit the change with the `git commit --amend` command.

    	$ git commit --amend 

    Git opens an editor containing your last commit message.

3. Adjust your last comment to reflect this new change.

	Here is an example:

        Added a new sentence per Anaud's suggestion	

        Signed-off-by: Mary Anthony <mary@docker.com>

        # Please enter the commit message for your changes. Lines starting
        # with '#' will be ignored, and an empty message aborts the commit.
        # On branch 11038-fix-rhel-link
        # Your branch is up-to-date with 'origin/11038-fix-rhel-link'.
        #
        # Changes to be committed:
        #		modified:   docs/sources/installation/mac.md
        #		modified:   docs/sources/installation/rhel.md

4. Force push the change to your branch.
   
   The command syntax is:

        git push -f origin <branch_name>

**Comment by draghuram** [comment]
@moxiegirl I agree that parentheticals are a problem so I have adapted your suggestions (with minor modifications). 

On a separate note, do we need to number the list items explicitly in Markdown? This is not convenient especially when items are inserted in between, necessitating re-numbering of the items that come after the one that is inserted.

**Comment by thaJeztah** [comment]
> On a separate note, do we need to number the list items explicitly in Markdown?

Markdown itself doesn't care about the actual numbers, but it *does* make the document easier to read when viewing the document as plain-text (which is, IMO a great feature of Markdown)

Anyway, I think this LGTM, thanks!

File: docs/sources/project/review-pr.md [hunk 2]
```diff
@@ -72,15 +79,15 @@ To update your existing pull request:
         #		modified:   docs/sources/installation/mac.md
         #		modified:   docs/sources/installation/rhel.md
 
-4. Push to your origin.
+5. Force push the change to your origin.
 
-        $ git push origin
+        $ git push -f origin <branch_name>
```
Comment (moxiegirl) [inline docs/sources/project/review-pr.md]: We use $ to indicate an actual command line.  That isn't the case here. This is a syntax example.  Can you remove that please?

File: docs/sources/project/review-pr.md [hunk 3]
```diff
@@ -49,15 +49,22 @@ need to update your pull request with additional changes.
...
-1. Change one or more files in your local `docker-fork` repository.
+1. Checkout the PR branch in your local `docker-fork` repository.  
 
-2. Commit the change with the `git commit --amend` command.
+    This is the branch associated with your request.
+
+2. Change one or more files and then stage your changes using the
+   command `git add`.
+
+    	$ git add <path_or_filename>
```
Comment (moxiegirl) [inline docs/sources/project/review-pr.md]: We use $ to indicate an actual command line.  That isn't the case here. This is a syntax example.  Can you remove that please?

**Comment by moxiegirl** [comment]
@draghuram I try to number in order as courtesy for the reasons mentioned by @thaJeztah.  It is also helpful when writing to know how many steps you've actually written.  There is some writing research around procedure length. Editorially, generally you should:

image

-- from http://amzn.com/0131428993

**Comment by draghuram** [comment]
I removed `$` from syntax examples. I also removed `git commit --amend` from the description as it seemed to be redundant. 

@thaJeztah @moxiegirl I guess I didn't make myself very clear regarding the numbering of list items. I totally agree that they should be numbered but I was hoping that the tool (Markdown here) will do that for me. Something on the lines of:
    http://docutils.sourceforge.net/docs/user/rst/quickref.html#enumerated-lists

@thaJeztah I too like to have documentation text files readable which will obviate the need to keep switching to the browser. I found that reference API files (`docker_remote_api*`), in particular, are not exactly easy to read what with very long lines and lack of blank lines. I hope to reformat the API docs at some point in order to make them more readable in text form.

**Comment by moxiegirl** [comment]
LGTM

**Comment by moxiegirl** [comment]
Ping @jamtur01 @SvenDowideit @fredlf one more LGTM from a maintainer please

**Comment by jamtur01** [comment]
LGTM

**Pull request merged**